### PR TITLE
move rubocop into foreman.yml

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: .github/workflows/foreman.yml [hunk 1]
```diff
@@ -11,8 +11,24 @@ env:
...
+      BUNDLE_WITHOUT: assets:console:development:dynflow_sidekiq:ec2:journald:jsonp:libvirt:openid:openstack:ovirt:redis:service:telemetry:vmware
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.7
+          bundler-cache: true
+      - name: Run rubocop
+        run: bundle exec rubocop -P
```
Comment (ekohl) [inline .github/workflows/foreman.yml:42]: When rubocop is >= 1.2 you can use `--format github` to get GH annotations, but we'd need to update theforeman-rubocop from 0.0.6 to 0.1.x. We should add that to the list of updates to perform: https://github.com/theforeman/theforeman-rubocop/pull/8#issuecomment-1825840790

**Review by ekohl — COMMENTED** [review]
Any thoughts on combining RuboCop with matrix building?

**Comment by evgeni** [comment]
> Any thoughts on combining RuboCop with matrix building?

You mean "in the same step"? I'd leave it separated for visibility reasons.

File: .github/workflows/foreman.yml (hunk 1 above)
Comment (evgeni) [reply .github/workflows/foreman.yml:42]: Yeah, updating tfm-r would be good.

**Comment by ekohl** [comment]
If you separate it out, would it make sense to reuse https://github.com/theforeman/actions/blob/v0/.github/workflows/rubocop.yml? And perhaps also run it after matrix building. So it's

* setup_matrix
* rubocop
* tests
* dummy

Then in rubocop you can use `matrix.ruby[0]` to get the Ruby version. If so, we should adopt the policy to always list the newest Ruby version first.

**Comment by evgeni** [comment]
Does the Ruby version we run Rubocop with really matter (as long it's a compatible one)?
(Remembering to keep the Rubies sorted will be hard, but I think it just doesn't matter which one we pick)

**Comment by ekohl** [comment]
Not a lot. The reuse of the matrix is more important than the exact order. Newest for RuboCop is IMHO preferred, but not required.

**Comment by evgeni** [comment]
There you go.

I briefly considered to run Rubocop for all rubies (like, you know, with a real matrix), but then could not find a good reason to do so.

**Review by ekohl — APPROVED** [review]

**Pull request merged**